modules/core/04-channel/keeper/grpc_query.go

```diff
@@ -398,18 +398,53 @@ func (q Keeper) UnreceivedPackets(c context.Context, req *types.QueryUnreceivedP
 
 	ctx := sdk.UnwrapSDKContext(c)
 
-	unreceivedSequences := []uint64{}
+	channel, found := q.GetChannel(sdk.UnwrapSDKContext(c), req.PortId, req.ChannelId)
+	if !found {
+		return nil, status.Error(
+			codes.NotFound,
+			errorsmod.Wrapf(types.ErrChannelNotFound, "port-id: %s, channel-id %s", req.PortId, req.ChannelId).Error(),
+		)
+	}
 
-	for i, seq := range req.PacketCommitmentSequences {
-		if seq == 0 {
-			return nil, status.Errorf(codes.InvalidArgument, "packet sequence %d cannot be 0", i)
+	unreceivedSequences := []uint64{}
```

Comment:
nit: I know this was already in place,, but I'm fairly sure we favour declaration over initialisation in many other parts of the codebase. Feel free to ignore, maybe we could create an issue to clean up most of these? 

```suggestion
	var unreceivedSequences []uint64
```

Reply:
Sure, will fix it up and then open an issue about this (this seems like something else the linter should be catching)

Reply:
ah, so this was probably initialized here since we check against it in the test file that uses a non-nil slice

Reply:
I'm fairly certain this was probably done to appease the tests. I think it's fine to change. This should work?
```diff
--- a/modules/core/04-channel/keeper/grpc_query.go
+++ b/modules/core/04-channel/keeper/grpc_query.go
@@ -406,7 +406,8 @@ func (q Keeper) UnreceivedPackets(c context.Context, req *types.QueryUnreceivedP
                )
        }
 
-       unreceivedSequences := []uint64{}
+       var unreceivedSequences []uint64
+
        switch channel.Ordering {
        case types.UNORDERED:
                for i, seq := range req.PacketCommitmentSequences {
diff --git a/modules/core/04-channel/keeper/grpc_query_test.go b/modules/core/04-channel/keeper/grpc_query_test.go
index 60e7012ef..6b0a414ed 100644
--- a/modules/core/04-channel/keeper/grpc_query_test.go
+++ b/modules/core/04-channel/keeper/grpc_query_test.go
@@ -1208,7 +1208,7 @@ func (suite *KeeperTestSuite) TestQueryUnreceivedPackets() {
 
                                suite.chainA.App.GetIBCKeeper().ChannelKeeper.SetPacketReceipt(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, 1)
 
-                               expSeq = []uint64{}
+                               expSeq = []uint64(nil)
                                req = &types.QueryUnreceivedPacketsRequest{
                                        PortId:                    path.EndpointA.ChannelConfig.PortID,
                                        ChannelId:                 path.EndpointA.ChannelID,

```

Reply:
opened #3376